algorithms/src/Kokkos_Random.hpp

```diff
@@ -728,6 +728,8 @@ struct Random_UniqueIndex<
 };
 #endif
 
+enum RandomPoolConstructorIsAsync : bool { no = false, yes = true };
+
```

Comment:
why a runtime and not a compile time selection?

Reply:
Because that would force us to use template and that feels unnecessary here

Reply:
ok